lib/kafka/async_producer.rb

```diff
@@ -77,23 +77,13 @@ def initialize(sync_producer:, max_queue_size: 1000, delivery_threshold: 0, deli
       @queue = Queue.new
       @max_queue_size = max_queue_size
 
-      @worker_thread = Thread.new do
-        worker = Worker.new(
-          queue: @queue,
-          producer: sync_producer,
-          delivery_threshold: delivery_threshold,
-        )
-
-        worker.run
-      end
-
-      @worker_thread.abort_on_exception = true
+      @worker = Worker.new(
+        queue: @queue,
+        producer: sync_producer,
+        delivery_threshold: delivery_threshold,
+      )
 
-      if delivery_interval > 0
-        Thread.new do
-          Timer.new(queue: @queue, interval: delivery_interval).run
-        end
-      end
+      @timer = Timer.new(queue: @queue, interval: delivery_interval)
```

Comment:
This will _always_ start the timer thread, even when it's not needed. I'm not sure if that's a performance problem.